src/components/BottomBar/BottomBarState.ts

```diff
@@ -24,6 +29,21 @@ export const reportCourseColorChange = (courseUniqueID: number, color: string):
   );
 };
 
+const getDetailedInformationForBottomBar = async (
+  roster: string,
+  subject: string,
+  number: string
+) => {
+  const courses: readonly CornellCourseRosterCourse[] = (
+    await fetch(
+      `https://classes.cornell.edu/api/2.0/search/classes.json?roster=${roster}&subject=${subject}`
+    ).then(response => response.json())
+  ).data.classes;
```

Comment:
I decide to fetch the data from cornell roster ondemand, because then we don't need to store those data on our own.

Reply:
Do you think we would ever run the risk of hitting the API too frequently once we launch? I want to make sure that this would be ok in the long term.

Reply:
No, this one use is the exactly the one we hit rarely. Bottom bar isn't open all the time, and course data is not refetched all the time.